sd-validation/helpers/walk-object.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * Recursively (DFS) traverses an object and calls provided function for each field.
+ *
+ * @param {*} obj
+ * @param {function(string, any, Array<string>, any): void} callback
+ * @param {Array<string>} path
+ */
+module.exports = function walkObject(obj, callback, path = []) {
+  if (obj === null) {
```

Comment:
do we care about guarding against non-objects passed in? we may not, but the type won't help prevent

Reply:
everything being passed in here is already `*` or has been filtered out for `null` already, so I don't think it matters